packages/react/src/components/UIShell/HeaderMenu.tsx

```diff
@@ -21,7 +21,29 @@ import { composeEventHandlers } from '../../tools/events';
  * with managing focus. It also passes along refs to each child so that it can
  * help manage focus state of its children.
  */
-class HeaderMenu extends React.Component {
+
+interface HeaderMenuProps {
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  className?: string;
+  focusRef?: React.Ref<any>;
+  isActive?: boolean;
+  isCurrentPage?: boolean;
+  menuLinkName: string;
+  onBlur?: (event: React.FocusEvent<HTMLLIElement>) => void;
+  onClick?: (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => void;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLLIElement>) => void;
+  renderMenuContent?: () => JSX.Element;
+  tabIndex?: number;
+  children?: React.ReactNode;
```

Comment:
Copying the jsdoc comments from proptypes to be up here as well would be ideal. They're [important for intellisense](https://code.visualstudio.com/docs/typescript/typescript-editing#_jsdoc-support)

Reply:
will add them from now